web/src/views/SubmissionPortal/HarmonizerView.vue

```diff
@@ -355,11 +355,19 @@ export default defineComponent({
           if (!template || !template.sampleDataSlot || !template.schemaClass) {
             return;
           }
+
+          // The spreadsheet has slot names as the header row. So `sheet_to_json` will produce array
+          // of objects with slot names as keys. But we want the imported data to be keyed on slot
+          // IDs. This code reads the worksheet data and remaps the keys from slot names to IDs.
+          const slotIdToNameMap = harmonizerApi.getHeaderRow(template.schemaClass);
+          const slotNameToIdMap = Object.fromEntries(Object.entries(slotIdToNameMap).map(([k, v]) => [v, k]));
+          const worksheet_data: Record<string, string>[] = utils.sheet_to_json(worksheet);
+          const remapped_data = worksheet_data.map((row) => Object.fromEntries(Object.entries(row)
+            .filter(([column]) => slotNameToIdMap[column] !== undefined)
```

Comment:
A question arose for me while reading this code in isolation, and without being familiar with the rest of the file: I infer, from the previous line, that `row` is not a scalar. Is `column` here a column (group of "cells") or a column name (scalar)? If the latter, I think I'd prefer the variable name `columnName`.

Reply:
Fixed in https://github.com/microbiomedata/nmdc-server/pull/1097/commits/50ca68278db99acd755c82d3584137f82c866365

Reply:
Thanks, good points. Even using "column" at all here is a little ambiguous here, so I've updated to use `slotName`. I also updated `worksheet_data` and `remapped_data` to be camel case. Whoops. Forgot I wasn't in Python anymore 🤦🏻